tests/test_pyrflx.py

```diff
@@ -0,0 +1,291 @@
+
+    def test_length_fields(self) -> None:
+        tlv = self.tlv.new()
+        tlv.set("Tag", "Msg_Data")
+        tlv.set("Length", 1)
+        self.assertEqual(tlv.accessible_fields, ["Tag", "Length", "Value", "Checksum"])
+        tlv.set("Value", b"\x01")
+        self.assertEqual(tlv.accessible_fields, ["Tag", "Length", "Value", "Checksum"])
+
+    def test_error_fields(self) -> None:
+        tlv = self.tlv.new()
+        tlv.set("Tag", "Msg_Error")
+        self.assertEqual(tlv.accessible_fields, ["Tag"])
+
+    def test_error_reset_fields(self) -> None:
+        tlv = self.tlv.new()
+        tlv.set("Tag", "Msg_Data")
+        tlv.set("Length", 1)
+        self.assertEqual(tlv.accessible_fields, ["Tag", "Length", "Value", "Checksum"])
+        tlv.set("Tag", "Msg_Error")
+        self.assertEqual(tlv.accessible_fields, ["Tag"])
+
+    def test_fields_complex(self) -> None:
+        tlv = self.tlv.new()
+        self.assertEqual(tlv.accessible_fields, ["Tag"])
+        tlv.set("Tag", "Msg_Error")
+        self.assertEqual(tlv.accessible_fields, ["Tag"])
+        tlv.set("Tag", "Msg_Data")
+        self.assertEqual(tlv.accessible_fields, ["Tag", "Length"])
+        tlv.set("Length", 1)
+        self.assertEqual(tlv.accessible_fields, ["Tag", "Length", "Value", "Checksum"])
+        tlv.set("Value", b"\x01")
+        self.assertEqual(tlv.accessible_fields, ["Tag", "Length", "Value", "Checksum"])
+        tlv.set("Checksum", 0xFFFFFFFF)
+        self.assertEqual(tlv.accessible_fields, ["Tag", "Length", "Value", "Checksum"])
+        tlv.set("Tag", "Msg_Error")
+        self.assertEqual(tlv.accessible_fields, ["Tag"])
+
+    def test_final(self) -> None:
+        tlv = self.tlv.new()
+        self.assertFalse(tlv.final)
+        tlv.set("Tag", "Msg_Error")
+        self.assertTrue(tlv.final)
+        tlv.set("Tag", "Msg_Data")
+        self.assertFalse(tlv.final)
+        tlv.set("Length", 1)
+        self.assertFalse(tlv.final)
+        tlv.set("Value", b"\x01")
+        self.assertFalse(tlv.final)
+        tlv.set("Checksum", 0xFFFFFFFF)
+        self.assertTrue(tlv.final)
+
+    def test_set_fields(self) -> None:
+        tlv = self.tlv.new()
+        self.assertEqual(tlv.set_fields, [])
+        tlv.set("Tag", "Msg_Data")
+        self.assertEqual(tlv.set_fields, ["Tag"])
+        tlv.set("Length", 1)
+        self.assertEqual(tlv.set_fields, ["Tag", "Length"])
+        tlv.set("Value", b"\x01")
+        self.assertEqual(tlv.set_fields, ["Tag", "Length", "Value"])
+        tlv.set("Checksum", 0xFFFFFFFF)
+        self.assertEqual(tlv.set_fields, ["Tag", "Length", "Value", "Checksum"])
+
+    def test_set_value(self) -> None:
+        tlv = self.tlv.new()
+        v1 = b"\x01\x02\x03\x04\x05\x06\x07\x08"
+        v2 = b"\x01\x02\x03\x04\x05\x06\x07\x08\x09\x10"
+        tlv.set("Tag", "Msg_Data")
+        tlv.set("Length", 8)
+        tlv.set("Value", v1)
+        with self.assertRaises(ValueError):
```

Comment:
I think it would be good to generally check also the error message using `assertRaisesRegex` (especially for reused built-in exceptions).

Reply:
Fixed by https://github.com/Componolit/RecordFlux/pull/158/commits/91889be32908f1f27383b00a59a97eca8d085b77.

Reply:
The `assertRaises` shown above still exists (although all others where replaced).

Also I do not like the unnecessary use of `.*` in some regex.

Reply:
Fixed by https://github.com/Componolit/RecordFlux/pull/158/commits/15de70bb6f279c45f218b2cb3f425da70a711cfa.